pandas/core/internals/construction.py

```diff
@@ -970,7 +970,17 @@ def convert(arr):
                     if dtype_backend != "numpy" and arr.dtype == np.dtype("O"):
                         new_dtype = StringDtype()
                         arr_cls = new_dtype.construct_array_type()
-                        arr = arr_cls._from_sequence(arr, dtype=new_dtype)
+                        try:
+                            # Addressing (#59242)
+                            # Byte data that could not be decoded into
+                            # a string would throw a UnicodeDecodeError exception
+
+                            # Try and greedily convert to string
+                            # Will fail if the object is bytes
+                            arr = arr_cls._from_sequence(arr, dtype=new_dtype)
```

Comment:
Shouldn't this ideally return `pandas.ArrowDtype(pyarrow.binary())` type?

Reply:
that makes sense thanks! so looks like the previous logic was not taking into account pyarrow types when doing this conversation so I've added logic similar to my initial change where we try to convert to a pyarrow string but then fall back to binary if we run into an invalid error (i.e. we tried to parse but it failed due to an encoding error). Please let me know what you think! Was also considering trying to type check the contents of `arr` to see if it has string or bytes data but seems like greedily trying to convert ends up being better performance in most cases (since we might have to search the whole arr to see if one of the elements is a bytes sequence that can't be converted to a string)